frontend/src/app/shared/components/storages/services/upload-storage-files.service.ts

```diff
@@ -0,0 +1,128 @@
+import { TimezoneService } from 'core-app/core/datetime/timezone.service';
+
+@Injectable()
+export class UploadStorageFilesService {
+  constructor(
+    private readonly httpClient:HttpClient,
+    private readonly timezoneService:TimezoneService,
+  ) {}
+
+  public uploadFile(uploadLink:IUploadLink, file:File):Observable<IStorageFile> {
+    const url = new URL(uploadLink._links.destination.href);
+    const token = url.username;
+    const password = url.password;
+    url.username = '';
+    url.password = '';
+
+    const headers = {
+      Authorization: `Basic ${btoa(`${token}:${password}`)}`,
+      'X-External-Request': 'true',
+    };
+
+    const method = uploadLink._links.destination.method;
+    return this.httpClient
+      .request(method, url.toString(), { body: file, headers })
+      .pipe(
+        switchMap(() => this.httpClient.request(
+          'propfind',
+          url.toString(),
+          {
+            body: this.propfindBody,
+            headers,
+            responseType: 'text',
+          },
+        )),
+        map((xml) => this.parseXmlResponse(xml)),
+      );
+  }
+
+  private parseXmlResponse(xml:string):IStorageFile {
+    const error = new Error(`Invalid response for uploaded file: ${xml}`);
+
+    const id = /<oc:fileid>(.*)<\/oc:fileid>/.exec(xml)?.pop();
+    if (!id) { throw error; }
+
+    const mimeType = /<d:getcontenttype>(.*)<\/d:getcontenttype>/.exec(xml)?.pop();
+    if (!mimeType) { throw error; }
+
+    const size = /<oc:size>(.*)<\/oc:size>/.exec(xml)?.pop();
+    if (!size) { throw error; }
+
+    const href = /<d:href>(.*)<\/d:href>/.exec(xml)?.pop();
+    const parts = href?.split('/');
+    if (!parts || parts.length < 1) { throw error; }
+
+    const name = parts.pop();
+    if (!name) { throw error; }
+
+    const location = `/${parts.slice(parts.indexOf('webdav') + 1).join('/')}`;
+
+    const date = /<d:getlastmodified>(.*)<\/d:getlastmodified>/.exec(xml)?.pop();
+    if (!date) { throw error; }
+    const createdAt = this.timezoneService.parseDatetime(date).toISOString();
+    const lastModifiedAt = createdAt;
+
+    const creator = /<oc:owner-display-name>(.*)<\/oc:owner-display-name>/.exec(xml)?.pop();
+    if (!creator) { throw error; }
+
+    return {
+      id,
+      name: decodeURIComponent(name),
+      location,
+      mimeType,
+      size: parseInt(size, 10),
+      createdAt,
+      createdByName: creator,
+      lastModifiedAt,
+      lastModifiedByName: creator,
+    };
+  }
+
+  private get propfindBody() {
+    return '<?xml version="1.0"?>\n'
+      + '<d:propfind xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">\n'
+      + '  <d:prop>\n'
+      + '    <oc:fileid />\n'
+      + '    <d:getlastmodified />\n'
+      + '    <d:getcontenttype />\n'
+      + '    <oc:size />\n'
+      + '    <oc:owner-display-name />\n'
+      + '  </d:prop>\n'
+      + '</d:propfind>';
```

Comment:
A template literal would make this a bit easier to handle

```suggestion
    return `
    	<?xml version="1.0"?>
        <d:propfind xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
          <d:prop>
            <oc:fileid />
            <d:getlastmodified />
            <d:getcontenttype />
            <oc:size />
            <oc:owner-display-name />
            </d:prop>
        </d:propfind>`;
```